platforms/windows/SQLitePlugin/SQLitePlugin.h

```diff
@@ -0,0 +1,143 @@
+    };
+
+    REACT_STRUCT(DBQuery);
+    struct DBQuery
+    {
+        REACT_FIELD(QID, L"qid");
+        int QID;
+
+        REACT_FIELD(Params, L"params");
+        JSValueArray Params; // optional
+
+        REACT_FIELD(SQL, L"sql");
+        std::string SQL;
+    };
+
+    REACT_STRUCT(EchoStringValueOptions);
+    struct EchoStringValueOptions
+    {
+        REACT_FIELD(Value, L"value");
+        std::string Value;
+    };
+
+    REACT_STRUCT(ExecuteSqlBatchOptions);
+    struct ExecuteSqlBatchOptions
+    {
+        REACT_FIELD(DBArgs, L"dbargs");
+        DBArgs DBArgs;
+        REACT_FIELD(Executes, L"executes");
+        std::vector<DBQuery> Executes;
+    };
+
+    REACT_STRUCT(OpenOptions);
+    struct OpenOptions
+    {
+        REACT_FIELD(Name, L"name");
+        // Path at which to store the database
+        std::string Name;
+
+        REACT_FIELD(AssetFileName, L"assetFilename");
+        // Optional. When creating the DB, uses this file as the initial state.
+        std::string AssetFileName;
+
+        REACT_FIELD(ReadOnly, L"readOnly");
+        bool ReadOnly;
+    };
+
+    REACT_MODULE(SQLitePlugin, L"SQLite");
+    struct SQLitePlugin : std::enable_shared_from_this<SQLitePlugin>
+    {
+    public:
+        REACT_METHOD(Attach, L"attach");
+        void Attach(
+            JSValue options,
+            std::function<void(std::string)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        REACT_METHOD(Close, L"close");
+        void Close(
+            CloseOptions options,
+            std::function<void(std::string)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        REACT_METHOD(DeleteDB, L"delete");
+        void DeleteDB(
+            DeleteOptions options,
+            std::function<void(std::string)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        REACT_METHOD(EchoStringValue, L"echoStringValue");
+        void EchoStringValue(
+            EchoStringValueOptions options,
+            std::function<void(std::string)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        REACT_METHOD(ExecuteSqlBatch, L"executeSqlBatch");
+        REACT_METHOD(ExecuteSqlBatch, L"backgroundExecuteSqlBatch");
+        void ExecuteSqlBatch(
+            ExecuteSqlBatchOptions options,
+            std::function<void(std::vector<JSValueObject>)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        REACT_METHOD(Open, L"open");
+        void Open(
+            OpenOptions options,
+            std::function<void(std::string)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        SQLitePlugin::SQLitePlugin();
+        SQLitePlugin::~SQLitePlugin();
+
+    private:
+        std::map<hstring, sqlite3*> OpenDBs;
+        ReactDispatcher SerialReactDispatcher{ ReactDispatcher::CreateSerialDispatcher() };

```

Comment:
probably a matter of preference, but how about changing it to lower-case to help indicate a member field vs method name?

Reply:
Updated